From: Eric Pouech <eric.pouech@gmail.com>
Subject: [PATCH] dbghelp: handling of complex location in dwarf
Message-Id: <162944854166.34236.14884258824479131601.stgit@euterpe>
Date: Fri, 20 Aug 2021 10:35:43 +0200

Don't recompute the location when we need to store a location block for
later recomputation.

It had also the bad side effect of letting the location to be absolute,
which wasn't expected at all.

Signed-off-by: Eric Pouech <eric.pouech@gmail.com>

---
 dlls/dbghelp/dwarf.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/dlls/dbghelp/dwarf.c b/dlls/dbghelp/dwarf.c
index 53274c4f597..a4f426ca6a0 100644
--- a/dlls/dbghelp/dwarf.c
+++ b/dlls/dbghelp/dwarf.c
@@ -930,7 +930,6 @@ static BOOL dwarf2_compute_location_attr(dwarf2_parse_context_t* ctx,
             *ptr = xloc.u.block.size;
             memcpy(ptr + 1, xloc.u.block.ptr, xloc.u.block.size);
             loc->offset = (ULONG_PTR)ptr;
-            compute_location(ctx->module, &lctx, loc, NULL, frame);
         }
     }
     return TRUE;

